docs/tutorials/TracingCode.adoc

```diff
@@ -0,0 +1,270 @@
+public CommandResult execute(String commandText) throws CommandException, ParseException {
+     logger.info("----------------[USER COMMAND][" + commandText + "]"); //Logging, safe to ignore
+
+     CommandResult commandResult;
+     Command command = addressBookParser.parseCommand(commandText); //Parse user input from String to a Command
+     commandResult = command.execute(model); //Executes the Command and stores the result
+
+     try {
+         //We can deduce that the previous line of code modifies model in some way since it's being stored here.
+         storage.saveAddressBook(model.getAddressBook());
+     } catch (IOException ioe) {
+         throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+     }
+
+     return commandResult;
+ }
+----
+. `LogicManager#execute()` appears to delegate most of the heavy lifting to other components. Let's take a closer
+look at each one.
+. `Step over` the logging code since it is of no interest to us now.
+image:StepOver.png[width=525px, height=83px]
++
+. `Step into` the line where user input in parsed from a String to a Command.
++
+.AddressBookParser#parseCommand()
+[source, java]
+----
+public Command parseCommand(String userInput) throws ParseException {
+    ...
+    final String commandWord = matcher.group("commandWord");
+    final String arguments = matcher.group("arguments");
+    ...
+----
+. `Step over` until you reach the `switch` statement. The `Variables` window now shows the value of both
+`commandWord` and `arguments` :
++
+image:Variables.png[width=1310px, height=403px]
+. We see that the value of `commandWord` is now `edit` but `arguments` is still not processed in any meaningful way.
+. Stepping into the `switch`, we obviously stop at
++
+.AddressBookParser#parseCommand()
+[source, java]
+----
+...
+case EditCommand.COMMAND_WORD:
+    return new EditCommandParser().parse(arguments);
+...
+----
+. Let's see what `EditCommandParser#parse()` does by stepping into it.
+. Stepping through the method shows that it calls `ArgumentTokenizer#tokenize()` and `ParserUtil#parseIndex()` to
+obtain the arguments and index required.
++
+[TIP]
+.Stepping out
+====
+Sometimes you might end up stepping into functions that are not of interest. Simply `step out` of them!
+====
+. The rest of the method seems to exhaustively check for the existence of each possible parameter of the `edit`
+command and store any possible changes in an `EditPersonDescriptor`. Recall that we can verify the contents of
+`editPersonDesciptor` through the `Variable` tool window.
++
+image:EditCommand.png[width=666px, height=313px]
+. Let's continue stepping through until we return to `LogicManager#execute()`.
++
+The sequence diagram below shows the details of the execution path through the Logic component.
+Does the execution path you traced in the code so far matches with the diagram?
++
+.Tracing an `edit` command through the Logic component
+image::LogicSequenceDiagram.png[width=1080px, height=476px]
```

Comment:
return arrows should be dashed

Reply:
Also, `[command: EditCommand]`

Reply:
If possible, show the full method call e.g., `parse(cmdString)`. Otherwise it is hard to distinguish between method names, parameters, return values.

Reply:
Return arrows dashed, named instance of EditCommand, changed function calls to use real parameter names.